src/magic_folder/magicpath.py

```diff
@@ -3,11 +3,40 @@
 
 from pyutil.assertutil import precondition, _assert
 
-def path2magic(path):
-    return re.sub(u'[/@]',  lambda m: {u'/': u'@_', u'@': u'@@'}[m.group(0)], path)
 
-def magic2path(path):
-    return re.sub(u'@[_@]', lambda m: {u'@_': u'/', u'@@': u'@'}[m.group(0)], path)
+def mangle_path_segments(segments):
+    """
+    Perform name-flattening on a list of path segments. Each element
+    represents one segment of a path (e.g. (sub)directory or the
+    actual filename) and is unicode.
+
+    The mangling consists of replacing path-separators with '@_' and
+    escaping '@' symbols inside a path-segment as '@@'.
+
+    :param list[unicode] segments: 1 or more unicode path segments
+
+    :returns unicode: flattened / mangled name
+    """
+    return u'@_'.join(
+        segment.replace(u'@', u'@@')
+        for segment in segments
+    )
+
+
+def unmangle_relative_path(mangled_path):
```

Comment:
Is there anything that makes this specific to relative paths?

Reply:
"mangled" paths should never be absolute

Reply:
I guess my thought here is that
1) the API "correctly" handles mangled absolute paths
2) there is nothing that enforces that the paths are relative.

so the name feels slightly misleading. I guess one solution would to verify that the path is relative before returning it?